src/csvcubed/utils/qb/validation/observations.py

```diff
@@ -180,13 +224,102 @@ def _validate_standard_shape_cube(
     return errors
 
 
+def _ensure_obs_val_col_linked(
+    column: QbColumn,
+    errors: list[ValidationError],
+    observed_column_title: Union[str, None],
+    defined_col_names: set[str],
+    obs_col_names: list[str],
+):
+
+    if observed_column_title is None:
+        errors.append(AttributeNotLinkedError(column.csv_column_title))
+    elif observed_column_title not in defined_col_names:
+        errors.append(
+            LinkedObsColumnDoesntExistError(
+                observed_column_title, column.csv_column_title
+            )
+        )
+    elif observed_column_title not in obs_col_names:
+        errors.append(
+            LinkedToNonObsColumnError(observed_column_title, column.csv_column_title)
+        )
+
+
 def _validate_pivoted_shape_cube(
-    cube: Cube
+    cube: Cube, obs_col_names: List[str]
 ) -> List[ValidationError]:
     errors: List[ValidationError] = []
 
     multi_measure_columns = get_columns_of_dsd_type(cube, QbMultiMeasureDimension)
-    if len(multi_measure_columns) > 0:
+    if any(multi_measure_columns):
+        """
+        In this case, the user has defined a redundant measure column.
+        All obs val columns already have their own measures declared.
+
+        Location, Average Income (meas defined), Average Age (meas defined), Measure
+        Birmingham, 22, 45.6, ???
+        """
+        errors.append(PivotedShapeMeasureColumnsExistError())
+
+    defined_col_names = {col.csv_column_title for col in cube.columns}
+
+    observed_value_columns = get_columns_of_dsd_type(cube, QbObservationValue)
```

Comment:
Have you made sure to merge `origin/pivoted-shape-feature` in here? I would definitely do it before merging. This will turn into `cube.get_columns_of_dsd_type(QbObservationValue)`.